frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java

```diff
@@ -0,0 +1,65 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import static com.virtuslab.gitmachete.frontend.actions.common.SyncToRemoteStatusDescriptionProvider.syncToRemoteStatusRelationToReadableBranchDescription;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.Presentation;
+import io.vavr.collection.List;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
+
+public interface ISyncToRemoteStatusDependentAction extends IBranchNameProvider, IExpectsKeyGitMacheteRepository {
+  String getActionName();
+
+  default String getDescriptionActionName() {
+    return getActionName();
+  }
+
+  List<SyncToRemoteStatus.Relation> getEligibleStatuses();
+
+  @UIEffect
+  default void syncToRemoteStatusDependentActionUpdate(AnActionEvent anActionEvent) {
+
+    Presentation presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    var branchName = getNameOfBranchUnderAction(anActionEvent);
+
+    if (branchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getDescriptionActionName() + "disabled due to undefined branch");
```

Comment:
Sadly string interpolation does not work within the interface's default :cry:

Reply:
https://github.com/antkorwin/better-strings/issues/15... maybe it will one day ;)